yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -168,8 +184,13 @@ export class PeerManager implements PeerManagerInterface {
     } else {
       this.logger.verbose(`Connected to transaction peer ${peerId.toString()}`);
     }
-    if (!this.config.p2pDisableStatusHandshake) {
+    if (this.config.p2pDisableStatusHandshake) {
+      return;
+    }
+    if (!this.config.p2pAllowOnlyValidators) {
```

Comment:
We issues either a status handshake or an auth handshake. Auth is a superset of status.